api/cron/takedown_nonprod_endpoints.py

```diff
@@ -0,0 +1,161 @@
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--region",
+        type=str,
+        default="us-west-1",
+        help=("Region to use when creating AWS session."),
+    )
+
+    # You cannot use `test_run` and `real_run` at the same time
+    feature_parser = parser.add_mutually_exclusive_group(required=False)
+    feature_parser.add_argument("--test_run", dest="test_run", action="store_true")
+    feature_parser.add_argument("--real_run", dest="test_run", action="store_false")
+    parser.set_defaults(test_run=True)
+    args = parser.parse_args()
+
+    return args
+
+
+def main():
+    args = parse_args()
+    print(args)
+    m = ModelModel()
+
+    if not args.test_run:
+        ops = input(
+            f"You've chosen to actually delete prod endpoints "
+            "(via flag `--real_run`),"
+            " are you sure? [y/n]"
+        )
+        if ops.lower() not in ("y", "yes"):
+            print(f"Aborting takedown script")
+            exit(1)
+
+    # Parse the DONT_TOUCH_MODELS
+    endpoints_name_not_touch = None
+    with open("cron/DONT_TOUCH_ENDPOINTS.txt") as f:
+        endpoints_name_not_touch = [str(t) for t in f.readlines()[0].split(",")]
+
+    # Make sure user is ok with current list of models not to touch
+    ops = input(
+        "You've chosen not to touch endpoints with names: "
+        + ", ".join(endpoints_name_not_touch)
+        + "Please make sure any additional endpoints you do not want delete are "
+        "added in cron/DONT_TOUCH_ENDPOINTS.txt. Continue? [y/n]"
+    )
+    if ops.lower() not in ("y", "yes"):
+        print(f"Aborting takedown script")
+        exit(1)
+
+    session = boto3.Session(
+        aws_access_key_id=build_config["aws_access_key_id"],
+        aws_secret_access_key=build_config["aws_secret_access_key"],
+        region_name=args.region,
+    )
+
+    sm = session.client("sagemaker")
+
+    # Retrieve all model endpoints
+    endpoint_response = sm.list_endpoints(
+        SortBy="Name",
+        SortOrder="Descending",
+        MaxResults=100,
+    )
+
+    endpoints = endpoint_response["Endpoints"]
+    while "NextToken" in endpoint_response:
+        next_token = endpoint_response["NextToken"]
+        endpoint_response = sm.list_endpoints(
+            SortBy="Name", SortOrder="Descending", MaxResults=100, NextToken=next_token
+        )
+
+        endpoints += endpoint_response["Endpoints"]
+
+    print(f"Total num of endpoints: {len(endpoints)}")
+
+    # Remove all DONT_TOUCH_ENDPOINTS from consideration
+    final_endpoints = []
+    for ep in endpoints:
+        ep_name = ep["EndpointName"]
+        if len(endpoints_name_not_touch) <= 0 or (
+            not any(
+                ep_name in bad_endpoint for bad_endpoint in endpoints_name_not_touch
+            )
+        ):
+            final_endpoints.append(ep)
+
+    print(
+        "Total num of endpoints after removing "
+        f"DONT_TOUCH_ENDPOINTS: {len(final_endpoints)}"
+    )
```

Comment:
Is this the total number of endpoints after endpoint removal? What about the endpoints that correspond to models in the prod db? We aren't touching those either.

Reply:
sorry the print is a little confusing, this is supposed to be the total number of endpoints we are considering for removal after removing DONT_TOUCH_ENDPOINTS.txt